File: SAE102/C/SAE102/utile.c
#include <stdio.h>
#include <stdlib.h>

int* meilleurCas(int n) {
    int* tab = (int*)malloc(n * sizeof(int));

    for (int i = 0; i < n; ++i) {
        tab[i] = i;
    }

    printf("Tableau avant le tri : ");
    for (int i = 0; i < n; ++i) {
        printf("%d ", tab[i]);
    }
    printf("\n");

    return tab;
}

int* pireCas(int n) {
    int* tab = (int*)malloc(n * sizeof(int));

    for (int i = 0; i < n; ++i) {
        tab[i] = n - i - 1;
    }

    printf("Tableau avant le tri : ");
    for (int i = 0; i < n; ++i) {
        printf("%d ", tab[i]);
    }
    printf("\n");

    return tab;
}

int* moyenCas(int n) {
    FILE* f = fopen("tableauMoyenCas.txt", "r");
    if (f == NULL) {
        perror("Erreur lors de l'ouverture du fichier");
        exit(EXIT_FAILURE);
    }

    int* tab = (int*)malloc(n * sizeof(int));

    for (int i = 0; i < n; ++i) {
        int content;
        fscanf(f, "%d", &content);
        tab[i] = content;
    }

    fclose(f);

    printf("Tableau avant le tri : ");
    for (int i = 0; i < n; ++i) {
        printf("%d ", tab[i]);
    }
    printf("\n");

    return tab;
}

File: SAE102/Python/utile.py
"meilleurCas = tableau déjà trié par ordre qu'on voulait (ici croissant)"
def meilleurCas(n):
  
    tab = []

    for i in range(n):

        tab.append(i)
    
    print("Tableau avant le tri :", tab)
    return tab

"pireCas = tableau déjà trié mais dans l'ordre inverse de ce qu'on veut (ici décroissant)"
def pireCas(n):
  
    tab = []

    for i in range(n):

        tab.append(n - i - 1)

    print("Tableau avant le tri :", tab)
    return tab

def moyenCas(n):
    f = open("tableauMoyenCas.txt", "r")

    tab = []

    for i in range(int(n)):
        content = f.readline()
        tab.append(int(content.split("\n")[0]))
    f.close()

    print("Tableau avant le tri :", tab)
    return tab